### DISCUSSION TITLE
List all matching Firebase dependencies versions in one place

### DISCUSSION THREAD

**Discussion by orestesgaolin** [firebase/flutterfire#12583] [Feature request] [answered]
When trying to find matching `firebase_core` and `firebase_auth` version I was quite confused to find that there's no clear indication which version of the package works with core package, and moreover there's no clear indication what version of the native Firebase library is used.

![CleanShot 2024-04-04 at 09 32 49](https://github.com/firebase/flutterfire/assets/16854239/730964c2-7b15-4565-bb6c-3fc9b11cfb1c)

What I propose is to present a table with following information:

- firebase_core version and release date
- matching firebase_xxx packages (e.g. as rows beneath the `firebase_core`)
- used native libraries version (or possible range)
- required Dart SDK version (should include transitive dependencies e.g. `web` requires Dart 3.3.0)
- for Flutter-dependent packages required Flutter version

The table should include at least some of the past versions of the Firebase libraries, so that it's possible to use Firebase SDK with Dart SDK 3.0.0 and newer.

Expected table may look something like this

|               	| version 	| release date 	| Firebase iOS 	| Firebase Android 	| Dart SDK 	| Flutter 	|
|---------------	|---------	|--------------	|--------------	|------------------	|----------	|---------	|
| firebase_core 	| 2.28.0  	| 2024-04-02   	| 10.22.0      	| 32.7.4           	| >=2.18.0 	| >=3.3.0 	|
| firebase_auth 	| 4.19.0  	|              	| ...          	| ...              	| ...      	| ...     	|
| ...           	| ...     	|              	| ...          	| ...              	| ...      	| ...     	|
| ----          	| ----    	|              	| ----         	| ----             	| ----     	| ----    	|
| firebase_core 	| ^2.27.0 	| 2024-03-08   	| 10.22.0      	| 32.7.3, 32.7.4   	| >=2.18.0 	| >=3.3.0 	|
| firebase_auth 	| ^4.17.8 	|              	| ...          	| ...              	| ...      	| ...     	|

**Comment by Lyokone** [comment]
Hello @orestesgaolin, this is something that we already have planned for 2024 and we are looking at ways of implementing it through the CLI, maybe something like 
```
$ flutterfire install 2.21.0
Select the dependencies you want to install
(X) core
(  ) auth
(  ) firestore 
...
(  ) app-check
```
So people can update all their dependencies in a single command.
What do yo think of this solution?

**Reply by orestesgaolin** [reply]
I'm happy for this to be exposed via cli, but having raw data source for this kind of information would make this much more accessible. For instance in my case I may have a firebase_auth dependency in one of the packages or plugins, and firebase_core or firebase_messaging in the main app. In such case I'd rather just go manually through pubspec.yaml, Podfile and gradle files instead of relying on cli.

To give you more context:
I doubt cli would help in my particular case, where I've migrated between older Flutter versions (3.13->3.16) and have used Firebase libraries from mid-2023. In my case I need to update Firebase to the earliest possible version that has [this fixed](https://github.com/firebase/flutterfire/issues/12442), but doesn't require newer Dart SDK (3.3.0)(which in fact is required [by transitive dependencies related to web](https://pub.dev/packages?q=dependency:web+firebase), it seems).

**Comment by orestesgaolin** [comment]
On a similar note, is there any way we could get the actual dependency version used in the generated changelog? I'm trying to figure out if I need to update firebase_crashlytics, and this is completely useless, sorry to say that...


![CleanShot 2024-05-06 at 10 15 01](https://github.com/firebase/flutterfire/assets/16854239/3484425e-de1f-494f-96e0-410c9f4e3635)

**Comment by TarekkMA** [comment, marked as answer]
@orestesgaolin, you can refer to this file: [VERSIONS.md](https://github.com/firebase/flutterfire/blob/master/VERSIONS.md), which includes all versions that belong to the same release.

**Reply by orestesgaolin** [reply]
Thanks! This seems helpful!

**Reply by orestesgaolin** [reply]
Wondering if there's any chance to improve the changelog content as well? This is super noisy and unhelpful right now

Either copying the changelog from the native library or just linking to the current version would be so much more useful

![CleanShot 2024-05-22 at 17 25 25](https://github.com/firebase/flutterfire/assets/16854239/eed68b18-9152-4d90-b686-27eb9de0d5e8)
